### PR TITLE
Wire in assertion that the appropriate product is created in Stripe

### PR DESCRIPTION
**Pull request by zspencer**
See: https://github.com/zinc-collective/compensated/issues/79

This is a big commit because it conflates the formatting changes from
https://github.com/zinc-collective/compensated/pull/89.

That said, it's also significant in it's own right. The ClientSandbox
now has the ability to query and assert against Stripe, so we will be
able to write full integration tests.

That said we _probably_ want to consider spinning out a fake Stripe API
so that our tests don't _always_ perform a full end-to-end check.

Perhaps in Coruru, if something like that doesn't exist already?

### CODE REVIEW

File: features/support/PaymentGateway.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  constructor({ type, secretKey }) {
+    this.type = type;
+    if(this.type === 'Stripe') {
+      this.stripe = Stripe(secretKey);
+    }
+  }
+
+  products() {
+    return new Promise((resolve, reject) => {
+      this.stripe.products.list({limit: 100}, (err, products) => {
```
Comment (user512) [inline features/support/PaymentGateway.js:14]: Is this `limit: 100` from Stripe?

Nvm, it's like we do `?limit=100` and the default is 10.
https://stripe.com/docs/api/products/list#list_products-limit

**Review by user512 — APPROVED** [review]

**Comment by user512** [comment]
I thought Coruro is used to test email, what do you have in mind in using Coruro to test product created in Stripe?

File: features/support/PaymentGateway.js (hunk 1 above)
Comment (zspencer) [reply features/support/PaymentGateway.js:14]: Yea, I figured 100 was a better default.

**Comment by zspencer** [comment]
Coruro's intent is to provide a seam for fake versions of external services. That can be email (in the case of coruro-mailcatcher) or it could be things like firebase or stripe or other such services.

**Pull request merged**